sympy/series/sequences.py

```diff
@@ -0,0 +1,522 @@
+from __future__ import print_function, division
+
+from sympy.core.expr import Expr
+from sympy.core.singleton import (S, Singleton)
+from sympy.core.compatibility import (range, integer_types, with_metaclass\
+                                      , is_sequence)
+from sympy.core.sympify import sympify
+from sympy.core.containers import Tuple
+from sympy.functions.elementary.integers import ceiling
+from sympy.utilities.misc import filldedent
+from sympy.sets.sets import Interval, Set
+
+
+class SeqBase(Expr):
+    """Base class for sequences"""
+
+    is_iterable = False
+
+    @property
+    def gen(self):
+        """Returns the generator for the sequence"""
+        return self._gen
+
+    @property
+    def _gen(self):
```

Comment:
I tried to follow the rule, never override a public method. Only private methods should be implemented. I find it cleaner. There is no need to rewrite docstrings. Though I agree, ``gen`` is a little vague. Maybe it's best to drop it altogether and use ``self.args[0]`` everywhere. It's just an other name of ``self.args[0]`` afterall.